src/components/shared/modal/Modal.tsx

```diff
@@ -19,96 +19,62 @@ export interface IModalState {
   open: boolean;
 }
 
-@autobind
-class Modal extends React.Component<IModalProps, IModalState> {
-  public static defaultProps: IModalProps = {
-    closeButton: false,
-    open: false,
-  };
-
-  public state: IModalState = {
-    open: this.props.open,
-  };
-
-  public componentDidMount() {
-    this.bodyOverflow(this.state.open);
-  }
-
-  public componentWillUnmount() {
-    this.bodyOverflow(false);
-  }
-
-  public UNSAFE_componentWillReceiveProps(nextProps: IModalProps) {
-    if (nextProps.open !== this.props.open) {
-      this.setState({ open: nextProps.open });
-      this.bodyOverflow(nextProps.open);
-    }
-  }
-
-  public render() {
-    const { title, closeButton } = this.props;
-    const showHeader = !!title || closeButton;
-
-    if (!this.state.open) {
-      return null;
-    }
-
-    return (
-      <Portal>
-        <React.Fragment>
-          <Overlay open={this.state.open} onClick={this.close} />
-
-          <div
-            className={classNames("modal", this.props.className)}
-            role="dialog"
-            aria-modal="true"
-            aria-labelledby="modal-title"
-          >
-            {showHeader && (
-              <div className="modal__header">
-                {title && (
-                  <span id="modal-title" className="modal__header__title">
-                    {title}
-                  </span>
-                )}
-
-                {closeButton && (
-                  <IconButton
-                    className="modal__header__close-button"
-                    iconName="close"
-                    onClick={this.close}
-                  />
-                )}
-              </div>
-            )}
-
-            {this.props.children}
-          </div>
-        </React.Fragment>
-      </Portal>
-    );
-  }
-
-  private bodyOverflow(open: boolean) {
+const Modal: FC<IModalProps> = ({
+  open,
+  title,
+  closeButton,
+  onClose,
+  className,
+  ...props
+}) => {
+  const showHeader = !!title || closeButton;
+
+  useEffect(() => {
     document.body.style.overflow = open ? "hidden" : "";
-  }
 
-  /** @note [refactor-consideration]: this function is never used */
-  private open() {
-    this.setState({ open: true });
+    return () => {
+      document.body.style.overflow = open ? "hidden" : "";
```

Comment:
From the old logic:

```ts
  public componentWillUnmount() {
    this.bodyOverflow(false);
  }
```

I think this could just be

```suggestion
      document.body.style.overflow = "";
```

Reply:
Agreed.